pkg/policy/policy_group_manager.go

```diff
@@ -183,6 +198,11 @@ func (m *policyGroupManager) UpdatePolicyGroup(ctx context.Context, policyGroup
 	if err != nil {
 		return nil, err
 	}
+
+	if currentPolicyGroup.Deleted {
```

Comment:
should we perform this check for `GetPolicyGroup` as well?

Reply:
I think it's useful to be able to fetch the group if there's still a reference to it. We allow `GetPolicy` to return deleted policies as well, so we may want to change that behavior if we decide to update `GetPolicyGroup`.

Reply:
that makes sense to me :+1: